Src/AutoFixture.xUnit.net.UnitTest/Scenario.cs

```diff
@@ -141,5 +141,85 @@ public void FavorArraysCausesArrayConstructorToBeInjectedWithFrozenItems([Frozen
         {
             Assert.Same(p1, p2);
         }
+
+        [Theory, AutoData]
+        public void FreezeFirstParameterShouldAssignSameInstanceToSecondParameter(
+            [Freeze]string p1,
+            string p2)
+        {
+            Assert.Same(p1, p2);
+        }
+
+        [Theory, AutoData]
+        public void FreezeFirstParameterByExactTypeShouldAssignSameInstanceToSecondParameter(
+            [Freeze(By = Matching.ExactType)]ConcreteType p1,
+            ConcreteType p2)
+        {
+            Assert.Same(p1, p2);
+        }
+
+        [Theory, AutoData]
+        public void FreezeFirstParameterByDirectBaseTypeShouldAssignSameInstanceToSecondParameter(
+            [Freeze(By = Matching.DirectBaseType)]ConcreteType p1,
+            AbstractType p2)
+        {
+            Assert.Same(p1, p2);
+        }
+
+        [Theory, AutoData]
+        public void FreezeFirstParameterByInterfaceShouldAssignSameInstanceToSecondParameter(
+            [Freeze(By = Matching.ImplementedInterfaces)]NoopInterfaceImplementer p1,
+            IInterface p2)
+        {
+            Assert.Same(p1, p2);
+        }
+
+        [Theory, AutoData]
+        public void FreezeFirstParameterByParameterNameShouldAssignSameInstanceToSecondParameter(
+            [Freeze(By = Matching.ParameterName, TargetName = "parameter")]string p1,

```

Comment:
I have to admit that having a convention is _much better_ than having to specify a member name as a `string` (it's almost reminiscent of [NMock](http://nmock.sourceforge.net)).

However, such a convention would introduce a couple of somewhat unexpected side effects:

- Renaming the parameter would likely break the test
- Renaming the member, in most cases, won't automatically rename the parameter, which would also break the affected tests. In this scenario, having the name as a `string` (with matching case) is more likely to get picked up by a refactoring tool or by a simple _find & replace_ operation.

In spite of this, I still prefer the convention approach that you suggested. However, I think we should document these edge cases.